templates/default/elasticsearch.init.erb

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env bash
+#
+# elasticsearch
+#
+<% if @platform_family == 'debian' %>
+### BEGIN INIT INFO
+# Provides:          elasticsearch
+# Required-Start:    $network $remote_fs $named
+# Required-Stop:     $network $remote_fs $named
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: Starts elasticsearch
+# Description:       Starts elasticsearch using start-stop-daemon
+### END INIT INFO
+<% else %>
+# chkconfig:   - 57 47
+# description: @service_name
+# processname: elasticsearch
+# config:      <%= @path_conf %>/elasticsearch.yml
+<% end %>
+
+# Source networking configuration
+#
+if [ -f /etc/sysconfig/network ]; then source /etc/sysconfig/network; fi
+
+# Set limits for environments ignoring `/etc/security/limits.d`
+#
+ulimit -n <%= @nofile_limit %>
+ulimit -l <%= @memlock_limit %>
+
+# Exit if networking is not up
+#
+[ "$NETWORKING" = "no" ] && exit
+
+PIDFILE='<%= @pid_file %>'
+ES_INCLUDE='<%= @path_conf %>/elasticsearch-env.sh'
+CHECK_PID_RUNNING=$(ps ax | grep 'java' | grep -e "es.pidfile=$PIDFILE" | sed 's/^\s*\([0-9]*\)\s.*/\1/')
+
+start() {
+    if [ -f $PIDFILE ]; then
+      # PIDFILE EXISTS -- ES RUNNING?
+      echo -e "PID file found in $PIDFILE"
+      es_pid="$(cat $PIDFILE)"
+      pid_running="$( ps ax | grep 'java' | grep $es_pid )"
+
+      if [ ! -z "$pid_running" ] ; then
+        # EXIT IF ES IS ALREADY RUNNING
+	      echo -e "\033[31;1mPID $es_pid still alive, Elasticsearch already running...\033[0m"
+	      return 1
+      fi
+    fi
+
+    echo -en "\033[1mStarting Elasticsearch...\033[0m"
+    touch $PIDFILE && chown <%= @user %> $PIDFILE
+    <% if @platform_family == 'debian' %>
```

Comment:
I always hated this: I think if we do provide an init script template, we need to put them to separate, platform specific directories?